log newly added construction and runtime stats at randomly selected iterations

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51394 log newly added construction and runtime stats at randomly selected iterations**
* #51386 add stats that can only be collected at runtime

log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

zhaojuanmao committed:
log newly added construction and runtime stats at randomly selected iterations

log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

SciPioneer commented on torch/lib/c10d/reducer.cpp:
```diff
@@ -988,6 +988,12 @@ void Reducer::prepare_for_forward() {
   std::lock_guard<std::mutex> lock(mutex_);
   num_iterations_++;
   forward_start_time_ = current_time_in_nanos();
+  // Log runtime (e.g. avg performance) stats at randomly selected iterations.
```
"10, 1000, or 10000" are not randomly selected iterations and can cause confusion. Could you please update the comment for the consistency?

zhaojuanmao replied:
sounds good!

rohan-varma commented on torch/lib/c10d/reducer.cpp:
```diff
@@ -988,6 +988,12 @@ void Reducer::prepare_for_forward() {
   num_iterations_++;
   forward_start_time_ = current_time_in_nanos();
+  // Log runtime (e.g. avg performance) stats at randomly selected iterations.
+  if (num_iterations_ == 10
```
Can we insert these iterations into a constant set so that they can be easily modified and extended as needed?

zhaojuanmao replied:
sounds good

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]

zhaojuanmao committed:
Update on "log newly added construction and runtime stats at randomly selected iterations"


log newly added construction and runtime stats at randomly selected iterations

Differential Revision: [D26161885](https://our.internmc.facebook.com/intern/diff/D26161885/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D26161885/)!

[ghstack-poisoned]
